File: app/src/main/java/com/dew/edward/roomword/viewmodel/WordViewModel.java
package com.dew.edward.roomword.viewmodel;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;

import com.dew.edward.roomword.data.Word;
import com.dew.edward.roomword.data.WordRepository;

import java.util.List;

/*
 * Created by Edward on 5/25/2018.
 */

public class WordViewModel extends AndroidViewModel {

    private WordRepository mRepository;

    private LiveData<List<Word>> mAllWords;

    public WordViewModel(@NonNull Application application) {
        super(application);
        this.mRepository = new WordRepository(application);
        this.mAllWords = mRepository.getAllWords();
    }

    public LiveData<List<Word>> getAllWords() {
        return mAllWords;
    }

    public void insert(Word word) {
        mRepository.insert(word);
    }
}

File: app/src/main/java/com/dew/edward/roomword/viewmodel/WordViewModel.kt
package com.dew.edward.roomword.viewmodel

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.arch.lifecycle.LiveData

import com.dew.edward.roomword.data.Word
import com.dew.edward.roomword.data.WordRepository

/*
 * Created by Edward on 5/25/2018.
 */

class WordViewModel(application: Application) : AndroidViewModel(application) {

    private val mRepository: WordRepository = WordRepository(application)

    val allWords: LiveData<List<Word>>

    init {
        this.allWords = mRepository.allWords
    }

    fun insert(word: Word) {
        mRepository.insert(word)
    }
}
